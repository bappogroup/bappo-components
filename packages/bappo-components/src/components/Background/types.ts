export type BackgroundProps = {
  children?: string;
  testID?: string;
};
