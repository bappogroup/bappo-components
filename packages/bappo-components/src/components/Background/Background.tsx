import * as React from 'react';

import { styled } from '../../apis/Style';
import View from '../../primitives/View';

type Props = {
  children?: string;
  testID?: string;
};

export default function Background({ children, testID }: Props = {}) {
  return (
    <StyledView selectable testID={testID}>
      {children}
    </StyledView>
  );
}

const StyledView = styled(View)`
  background: #f7f7f7;
  padding: 16px;
`;
