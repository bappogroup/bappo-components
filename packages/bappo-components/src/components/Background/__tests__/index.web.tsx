import 'jest-styled-components';
import 'react-testing-library/cleanup-after-each';

import React from 'react';
import { render } from 'react-testing-library';

import Background from '..';

test('should render empty background', () => {
  const { getByTestId } = render(<Background testID="test"></Background>);
  expect(getByTestId('test')).toMatchSnapshot();
});
