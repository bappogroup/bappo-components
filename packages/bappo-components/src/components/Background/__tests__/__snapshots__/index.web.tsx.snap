// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render empty background 1`] = `
.c0 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  border-color: #191E26;
  border-style: solid;
  border-width: 0;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  position: relative;
  min-height: 0;
  min-width: 0;
}

.c1 {
  background: #f7f7f7;
  padding: 16px;
}

<div
  class="c0 c1"
  data-testid="test"
/>
`;
